test/system_test.py

```diff
@@ -39,7 +40,11 @@ def spack_install_and_test(spec: str,
     If log_filename is None, spec is used to create one.
     """
 
-    log_filename = sanitized_filename(log_filename or spec)
+    func_name = inspect.currentframe().f_back.f_code.co_name.replace(
+        'test_', '')
+    class_name = inspect.currentframe().f_back.f_locals.get(
+        'self', None).__class__.__name__.replace('Test', '')
+    log_filename = sanitized_filename(class_name + '-' + func_name)
```

Comment:
This overwrites the function argument log_filename.

Reply:
Would the following solve the issue?
```
if log_filename:
   log_filename = sanitized_filename(class_name + '-' + func_name)
```